control-plane-operator/hostedclusterconfigoperator/controllers/resources/alerts/assets/apiusage.yaml

```diff
@@ -0,0 +1,36 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: api-usage
+  namespace: openshift-kube-apiserver
+spec:
+  groups:
+    - name: pre-release-lifecycle
+      rules:
+        - alert: APIRemovedInNextReleaseInUse
+          annotations:
+            message: >-
+              Deprecated API that will be removed in the next version is being used. Removing the workload that is using
+              the {{ $labels.group }}.{{ $labels.version }}/{{ $labels.resource }} API might be necessary for
+              a successful upgrade to the next cluster version.
+              Refer to `oc get apirequestcounts {{ $labels.resource }}.{{ $labels.version }}.{{ $labels.group }} -o yaml` to identify the workload.
+          expr: |
+            group(apiserver_requested_deprecated_apis{removed_release="1.23"}) by (group,version,resource) and (sum by(group,version,resource) (rate(apiserver_request_total{system_client!="kube-controller-manager",system_client!="cluster-policy-controller"}[4h]))) > 0
```

Comment:
am I understanding correctly this is reconciling these alerts in the guest cluster? are these metrics available there? cc @csrwng

Reply:
> am I understanding correctly this is reconciling these alerts in the guest cluster? are these metrics available there? cc @csrwng

Yes, the api-usage rules need to be reconciles inside the guest cluster. @enxebre please clarify what metrics availability is questioned?

Reply:
@enxebre yes the metrics should be available there because we add a service monitor to scrape the kube apiserver (default/kubernetes svc)